dns-query/src/adapter.ts

```diff
@@ -0,0 +1,59 @@
+import { DNSResponse, Execute } from '@chainlink/types'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+
+export enum DNSProviders {
+  Cloudfare = 'cloudfare',
+  Google = 'google',
+}
+
+export const endpoints: Record<string, string> = {
+  [DNSProviders.Cloudfare]: 'https://cloudflare-dns.com/dns-query',
+  [DNSProviders.Google]: 'https://dns.google/dns-query',
+}
+
+export const inputParams = {
+  name: true,
+  type: true,
+  do: false,
+  cd: false,
+}
+
+export const execute: Execute = async (input) => {
+  try {
+    const validator = new Validator(input, inputParams)
+    if (validator.error) throw validator.error
+
+    const dnsProvider = util.getRequiredEnv('DNS_PROVIDER')
+    if (!Object.values(DNSProviders).includes(dnsProvider))
+      throw new Error(`Unknown DNS Provider: ${dnsProvider}`)
```

Comment:
Please require all configuration to be given to you instead of reading it yourself from the ENV.

```suggestion
const execute: ExecuteWithConfig = async (input, config) => {
  try {
    const validator = new Validator(input, inputParams)
    if (validator.error) throw validator.error
```

Then you can export a `makeConfig` fn that will help create this config by reading from the ENV:

```javascript
export const makeConfig = () => {
    const provider = util.getRequiredEnv('DNS_PROVIDER')
    if (!Object.values(DNSProviders).includes(provider))
      throw new Error(`Unknown DNS Provider: ${provider}`)
    return { provider }
}

Reply:
We could also add an `endpoint/url` as configuration, so we have a few predefined providers but also anyone can set up an adapter with custom `endpoint/url` without needing to update the code.

Reply:
Do we want that to be possible? They could then call any endpoint, even if it's not a DNS one

Reply:
Yeah, why not enable that? If they provide a non-compatible endpoint their adapter won't work. I am ok with that.